pipx/animate.py

```diff
@@ -59,12 +60,17 @@ def print_animation(
     period: float,
     animate_at_beginning_of_line: bool,
 ):
+    (term_cols, _) = shutil.get_terminal_size(fallback=(9999, 24))
     while not event.wait(0):
         for s in symbols:
             if animate_at_beginning_of_line:
-                cur_line = f"{s} {message}"
+                max_message_len = term_cols - len(f"{s} ... ")
+                cur_line = f"{s} {message:.{max_message_len}}"
+                if len(message) > max_message_len:
+                    cur_line += "..."
```

Comment:
your calculations are made with a space before the dots, shouldn't you append here the space before the dots?

Reply:
The extra text is `f"{s} "` (one space after {s}) before the message, and `"... "` (with extra space after dots) after the message.  Hence `f"{s} ... "`.

The extra space after the dots is because the last character on the terminal line didn't show up after printing the `\r`, so I made the maximum line one character less.